apps/examples/araplayer/AudioPDStream.cpp

```diff
@@ -0,0 +1,192 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include "AudioPDStream.h"
+#include "MediaPlayerWrapper.h"
+
+AudioPDStream::AudioPDStream()
+{
+}
+
+AudioPDStream::~AudioPDStream()
+{
+	removeMapAll();
+}
+
+int AudioPDStream::startStream()
+{
+	AudioHttpDownloader *pHttpDownloader;
+	int ret = -1;
+
+	pHttpDownloader = new AudioHttpDownloader();
+	if (pHttpDownloader == NULL) {
+		DEBUG_TRACE(LOG_ERR, "MultiRoomHttpDownloader error!");
+		return -1;
+	}
+
+	pHttpDownloader->mStream = this;
+	ret = pHttpDownloader->startHttpDownload(mDownloaderMap, mState->url, parseHttpData);
+	if (ret < 0) {
+		DEBUG_TRACE(LOG_ERR, "startHttpDownload error!");
+		return -1;
+	}
+
+	return ret;
+}
+
+int AudioPDStream::pauseStream()
+{
+	AudioHttpDownloader *pHttpDownloader;
+	int ret = -1;
+
+	for (mIter = mDownloaderMap.begin(); mIter != mDownloaderMap.end(); mIter++) {
```

Comment:
hmm, it allows only one stream per one context? or allows multi streams from one context?
This code is pausing all downloader in downloaderMap, actually?